src/store/setToolMode.js

```diff
@@ -8,67 +8,137 @@ import { state } from './../store/index.js';
  * respond to user input, and can create new data
  *
  * @export
- * @param {*} element
- * @param {*} toolName
- * @param {*} options
+ * @param {object} element
+ * @param {string} toolName
+ * @param {(object|number)} options
+ * @param {boolean} isTouchActive
  * @returns
  */
-const setToolActiveForElement = setToolModeForElement.bind(
-  null,
-  'active',
-  null
-);
-const setToolActive = setToolMode.bind(null, 'active', null);
+const setToolActiveForElement = function (
+  element,
+  toolName,
+  options,
+  isTouchActive
+) {
+  const tool = getToolForElement(element, toolName);
+
+  if (tool) {
+    _setActiveToolWithMatchingMouseButtonMaskToPassive(tool, element, options);
+    _setActiveToolWithIsTouchActiveToFalse(
```

Comment:
Both of these function names were very difficult to parse for me, I had to look at the function to get what was trying to be done. I think its because we are in `setToolActiveForElement`, and now we are talking about the _previously_ active tool straight away (if I understand correctly). I reckon this either needs a lot of comments, or lines 23 to 36 refactored to another function such as `resolveInputConflicts()` or something.

Not sure on the best solution, feel free to bounce back with a better idea.